pkg/geo/geos/geos.cc

```diff
@@ -1160,6 +1189,126 @@ CR_GEOS_Status CR_GEOS_HausdorffDistanceDensify(CR_GEOS* lib, CR_GEOS_Slice a, C
   return toGEOSString(error.data(), error.length());
 }
 
+//
+// PreparedGeometry
+//
+
+CR_GEOS_Status CR_GEOS_PrepareGeometry(CR_GEOS* lib, CR_GEOS_Slice a, CR_GEOS_PreparedGeometry* ret) {
+  std::string error;
+  auto handle = initHandleWithErrorBuffer(lib, &error);
+
+  auto wkbReader = lib->GEOSWKBReader_create_r(handle);
+  auto geom = lib->GEOSWKBReader_read_r(handle, wkbReader, a.data, a.len);
+  if (geom != nullptr) {
+    *ret = lib->GEOSPrepare_r(handle, geom);
+    // TODO: make sure underlying geom is freed too.
+    // lib->GEOSGeom_destroy_r(handle, geom);
```

Comment:
this needs to return a struct, so that we free the `GEOSGeometry` underneath.

Consider something like:

```
struct PreparedGeometry {
  CR_GEOS_Geometry *g;
   CR_GEOS_PreparedInternalGeometry *p; // this corresponds to PreparedGeometry in GEOS.
}

CR_GEOS_Status CR_GEOS_PrepareGeometry(CR_GEOS* lib, CR_GEOS_Slice a, CR_GEOS_PreparedGeometry* ret) {
    ....
    *ret = {.g = geom, .p = lib->GEOSPrepare_r(handle, geom)}
}
```

So that in `CR_GEOS_PreparedGeometryDestroy`, we call `GEOSGeom_destroy_r` on `g` as well.

Reply:
Ok, I changed the `PreparedGeometry` structure accordingly.
Please just note that in the updated Go code `geos.PrepareGeometry` on error returns an empty `PreparedGeometry` instead of `nil`, I hope I got it right.